src/linter.coffee

```diff
@@ -1,38 +1,44 @@
-_ = require 'lodash'
-async = require 'async'
-asyncHandlers = require 'async-handlers'
-fsExtra = require 'fs-extra'
-DependencyLinter = require './linter/dependency_linter'
-UsedModuleFinder = require './linter/used_module_finder'
-path = require 'path'
-
-
 class Linter
 
 
-  constructor: (@dir, {allowUnused, devFilePatterns, devScripts, ignoreFilePatterns}) ->
-    @dependencyLinter = new DependencyLinter {allowUnused, devFilePatterns, devScripts}
-    @usedModuleFinder = new UsedModuleFinder {@dir, ignoreFilePatterns}
+  constructor: (@dir, {@allowUnused, @devFilePatterns, @devScripts, @ignoreFilePatterns}) ->
 
 
-  lint: (done) ->
-    async.parallel {
-      listedModules: @getListedModules
-      usedModules: @usedModuleFinder.find
-    }, asyncHandlers.transform(@dependencyLinter.lint, done)
+  lint: (done) =>
+    async = require 'async'
+    asyncHandlers = require 'async-handlers'
```

Comment:
Another disadvantage of inlining the requires here is that it artificially increases the cyclomatic complexity score for methods, and makes them longer and harder to read in general. And it makes using external libraries more cumbersome. If I have to require `async` each time I want to run one of the many async methods, that can quickly add a lot of boilerplate to a code base (like 10-20% of total lines, if we use a lot of external libraries, as we should), and make me not want to use async or other libraries that much anymore.